File: 프로그래머스/lv2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    phone_book.sort()
    N = len(phone_book)

    for i in range(N-1):
        for j in range(i+1, N):
            l = len(phone_book[i])
            if phone_book[j][:l] == phone_book[i]:
                return False
            else:
                break
    return True

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        Arrays.sort(phone_book);
        for (int i=0;i<phone_book.length-1;i++) {
            String prefix = phone_book[i];
            String phone = phone_book[i+1];
            if (prefix.length() > phone.length()) {
                continue;
            }
            if (prefix.equals(phone.substring(0, prefix.length()))) {
                return false;
            }
        }
        return true;
    }
}